.github/workflows/test.yml

```diff
@@ -61,3 +61,10 @@ jobs:
         if: steps.nix-store-cache.outputs.cache-hit != 'true'
         run: |
           nix copy --to "file:///tmp/nix-cache?compression=zstd&parallel-compression=true" .#native-cargo-artifacts
+
+      - uses: actions/cache/save@v3
+        # Write a cache entry even if the tests fail but don't create any for the merge queue.
+        if: ${{ always() && github.event_name != 'merge_group' }}
+        with:
+          path: ${{ steps.restore-cache.inputs.path }}
+          key: ${{ steps.restore-cache.outputs.cache-primary-key }}
```

Comment:
What happens if the `file:///tmp/nix-cache` doesn't exist, such that there was a cache-hit so nothing was exported from nix? Does this fail?

Reply:
![image](https://github.com/noir-lang/noir/assets/15848336/e4392a8c-88d6-4ce1-b9b5-566856f498a7)